src/test/operators/join_test_runner.cpp

```diff
@@ -0,0 +1,590 @@
+    // Use both Equals and LessThan to trigger sorting/non-sorting mode of JSM's ColumnMaterializer.
+    for (const auto encoding_type : all_encoding_types) {
+      for (const auto data_type : all_data_types) {
+        for (const auto nullable : {false, true}) {
+          for (const auto table_type : all_input_table_types) {
+            for (const auto predicate_condition : {PredicateCondition::Equals, PredicateCondition::LessThan}) {
+              auto join_test_configuration = default_configuration;
+              join_test_configuration.input_left.encoding_type = encoding_type;
+              join_test_configuration.input_left.table_type = table_type;
+              join_test_configuration.data_type_left = data_type;
+              join_test_configuration.nullable_left = nullable;
+              join_test_configuration.input_right.encoding_type = encoding_type;
+              join_test_configuration.input_right.table_type = table_type;
+              join_test_configuration.data_type_right = data_type;
+              join_test_configuration.nullable_right = nullable;
+              join_test_configuration.predicate_condition = predicate_condition;
+
+              add_configuration_if_supported(join_test_configuration);
+            }
+          }
+        }
+      }
+    }
+
+    // JoinHash specific configurations varying the number of radix_bits.
+    // The number of radix bits affects the partitioning inside the JoinHash operator, thus table/chunk sizes and
+    // different join modes need to be tested in combination, since they all use this partitioning.
+    if constexpr (std::is_same_v<JoinOperator, JoinHash>) {
+      for (const auto radix_bits : {0, 1, 2, 5}) {
+        for (const auto join_mode : {JoinMode::Inner, JoinMode::Right, JoinMode::Semi}) {
+          for (const auto left_table_size : all_table_sizes) {
+            for (const auto right_table_size : all_table_sizes) {
+              for (const auto chunk_size : all_chunk_sizes) {
+                for (const auto predicate_condition : {PredicateCondition::Equals, PredicateCondition::NotEquals}) {
+                  auto join_test_configuration = default_configuration;
+                  join_test_configuration.join_mode = join_mode;
+                  join_test_configuration.predicate_condition = predicate_condition;
+                  join_test_configuration.input_left.table_size = left_table_size;
+                  join_test_configuration.input_left.chunk_size = chunk_size;
+                  join_test_configuration.input_right.table_size = right_table_size;
+                  join_test_configuration.input_right.chunk_size = chunk_size;
+                  join_test_configuration.radix_bits = radix_bits;
+
+                  add_configuration_if_supported(join_test_configuration);
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+
+    return configurations;
```

Comment:
std::sort(std::unique(configurations))???

Reply:
You mean the beauty that is this?

```c++
    std::sort(configurations.begin(), configurations.end());
    configurations.erase(std::unique(configurations.begin(), configurations.end()), configurations.end());
```

But yeah, helps:

Before: 1115
After: 1003

Before: 827
After: 459

Before: 740
After: 682

Before: 653
After: 562

Before: 272
After: 233

Reply:
Beautiful. :)